### PR TITLE
catboost: 1.2.5 -> 1.2.7

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Things done

Changelog: https://github.com/catboost/catboost/releases/tag/v1.2.7

- **catboost: move to by-name**
- **catboost: format**
- **catboost: 1.2.5 -> 1.2.7**

cc @PlushBeaver @natsukium

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by zakame** [comment]
Result of `nixpkgs-review pr 347888` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>catboost</li>
    <li>catboost.dev</li>
    <li>python311Packages.catboost</li>
    <li>python311Packages.catboost.dist</li>
    <li>python311Packages.orange3</li>
    <li>python311Packages.orange3.dist</li>
    <li>python312Packages.catboost</li>
    <li>python312Packages.catboost.dist</li>
    <li>python312Packages.orange3</li>
    <li>python312Packages.orange3.dist</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -19076,10 +19076,10 @@ with pkgs;
...
   captive-browser = callPackage ../applications/networking/browsers/captive-browser { };
 
-  catboost = callPackage ../development/libraries/catboost {
-    # https://github.com/catboost/catboost/issues/2540
-    cudaPackages = cudaPackages_11;
-  };
+  # catboost = callPackage ../development/libraries/catboost {
+  #   # https://github.com/catboost/catboost/issues/2540
+  #   cudaPackages = cudaPackages_11;
+  # };
```
Comment (zakame) [inline pkgs/top-level/all-packages.nix:19082]: Do these need to be here still, or can be removed entirely?

**Review by zakame — CHANGES_REQUESTED** [review]
Thanks for the update! 🙇 Just one small fix maybe 🙏

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (GaetanLepage) [reply pkgs/top-level/all-packages.nix]: I was testing if it was possible to remove it. But unfortunately no, we have to pin cuda to version 11.

**Comment by GaetanLepage** [comment]
> Thanks for the update! 🙇 Just one small fix maybe 🙏

No prob ! I have done this update hoping to fix the package for `x86_64-darwin`. I am not yet sure that it now builds.

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 347888`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 10 packages built:</summary>
  <ul>
    <li>catboost</li>
    <li>catboost.dev</li>
    <li>python311Packages.catboost</li>
    <li>python311Packages.catboost.dist</li>
    <li>python311Packages.orange3</li>
    <li>python311Packages.orange3.dist</li>
    <li>python312Packages.catboost</li>
    <li>python312Packages.catboost.dist</li>
    <li>python312Packages.orange3</li>
    <li>python312Packages.orange3.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 10 packages built:</summary>
  <ul>
    <li>catboost</li>
    <li>catboost.dev</li>
    <li>python311Packages.catboost</li>
    <li>python311Packages.catboost.dist</li>
    <li>python311Packages.orange3</li>
    <li>python311Packages.orange3.dist</li>
    <li>python312Packages.catboost</li>
    <li>python312Packages.catboost.dist</li>
    <li>python312Packages.orange3</li>
    <li>python312Packages.orange3.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 10 packages marked as broken and skipped:</summary>
  <ul>
    <li>catboost</li>
    <li>catboost.dev</li>
    <li>python311Packages.catboost</li>
    <li>python311Packages.catboost.dist</li>
    <li>python311Packages.orange3</li>
    <li>python311Packages.orange3.dist</li>
    <li>python312Packages.catboost</li>
    <li>python312Packages.catboost.dist</li>
    <li>python312Packages.orange3</li>
    <li>python312Packages.orange3.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.orange3</li>
    <li>python311Packages.orange3.dist</li>
    <li>python312Packages.orange3</li>
    <li>python312Packages.orange3.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>catboost</li>
    <li>catboost.dev</li>
    <li>python311Packages.catboost</li>
    <li>python311Packages.catboost.dist</li>
    <li>python312Packages.catboost</li>
    <li>python312Packages.catboost.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
@zakame I addressed your point. Any additional feedback ?

**Review by zakame — APPROVED** [review]

**Pull request merged**